source/elements/oneDAL/dalapi/generator.py

```diff
@@ -0,0 +1,42 @@
+from typing import (List, Text)
+
+class RstBuilder(object):
+    def __init__(self):
+        self._rst_list = []
+
+    def add_class(self, namespace: str, class_declaration: str, level=0):
+        self(f'.. cpp:namespace:: {namespace}', level)
+        self()
+        self(f'.. cpp:class:: {class_declaration}', level)
+        self()
+
+    def add_function(self, definition: str, level=0):
+        self(f'.. cpp:function:: {definition}', level)
+        self()
+
+    def add_property(self, definition: str, level=0):
+        self(f'.. cpp:member:: {definition}', level)
+        self()
+
+    def add_doc(self, description: str, level=0):
+        if len(description) > 0:
+            self(self._format_description(description), level)
+            self()
+
+    def add_code_block(self, listing: List[Text], level=0):
+        self(f'.. code-block:: cpp', level)
+        self()
+        for line in listing:
+            self(line, level + 1)
+        self()
+
+    def __call__(self, string: str = '', level:int = 0):
+        self._rst_list.append(' ' * level * 3 + string)
+
+    def build(self):
+        return self._rst_list
+
+    def _format_description(self, description):
+        if not description.endswith('.'):
+            return description + '.'
```

Comment:
I'm not sure about this. You get things like `The predicted labels.` or `The trained k-NN model.`, and they are not full sentences, there should not be a full stop at the end of these.

Reply:
Do you suggest to remove `.` or use full sentences?

Reply:
BTW, there is a full stop in DAL docs for such descriptions. http://oneapi-src.github.io/oneDAL/dev_guide/algorithms/k_nearest_neighbors/k-nearest-neighbors-knn-classifier.html#training

Reply:
I'd remove `.`

> BTW, there is a full stop in DAL docs for such descriptions. http://oneapi-src.github.io/oneDAL/dev_guide/algorithms/k_nearest_neighbors/k-nearest-neighbors-knn-classifier.html#training

it's a bit different: we do have full sentences there sometimes, so we get a mix of two types of descriptions in the same table and it requires punctuation. But here I don't see the full sentences at all, so no need to punctuate.

Reply:
But it would be a mix of sentence and non sentence in general case.  What if I need to add more detailed description for some property or add a method?

Reply:
if it's probable, then let's leave the full stops.